### PR TITLE
Added explanations to School Form

### PR DESCRIPTION
**Pull request by GeorgebigG**
For the abbrev, I added an saying that abbreviations should be all lowercase. I added a term regex [WSMF]\d\d for UCSB since our terms are for example: F24, W25, S25, M25. And I changed the term description.

https://organic-george.dokku-09.cs.ucsb.edu/

### CODE REVIEW

**Review by crosscountrycoder — COMMENTED** [review]
Looks good to me, except please fix the errors in the testing (unit tests, coverage, stryker).

**Review by crosscountrycoder — COMMENTED** [review]
Looks good to me! Errors have been fixed.

**Review by JUstISAGE — DISMISSED** [review]
lgtm

File: frontend/src/main/components/School/SchoolForm.js [hunk 1]
```diff
@@ -56,16 +57,16 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
...
                             data-testid="SchoolForm-termRegex"
                             id="termRegex"
                             type="text"
                             isInvalid={Boolean(errors.termRegex)}
-                            {...register("termRegex", { required: true })}
+                            {...register("termRegex", { required: true, pattern: term_regex })}
                         />
                         <Form.Control.Feedback type="invalid">
-                            {errors.termRegex && 'Term Regex is required. '}
+                            {errors.termRegex && 'Term Regex has to include a single character for a Quarter and 2 digits for the year.'}
```
Comment (pconrad) [inline frontend/src/main/components/School/SchoolForm.js]:
```suggestion
                            {errors.termRegex && 'Term regex is required.'}
```

File: frontend/src/main/components/School/SchoolForm.js [hunk 2]
```diff
@@ -56,16 +57,16 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
...
                     <Form.Group className="mb-3" >
-                        <Form.Label htmlFor="termRegex">Term Regex</Form.Label>
+                        <Form.Label htmlFor="termRegex">Term Regex(Example: W24)</Form.Label>
                         <Form.Control
                             data-testid="SchoolForm-termRegex"
                             id="termRegex"
                             type="text"
                             isInvalid={Boolean(errors.termRegex)}
-                            {...register("termRegex", { required: true })}
+                            {...register("termRegex", { required: true, pattern: term_regex })}
```
Comment (pconrad) [inline frontend/src/main/components/School/SchoolForm.js]:
```suggestion
                            {...register("termRegex", { required: true})}
```

This is applying the regular expression `const term_regex = /[WSMF]\d\d/;` to the input for this field, which completely misunderstands what is supposed to be happening here.

Instead, we want the user to *input the regular expression against which the terms for their school will be matched*.

That is, if your schools has terms likes F24, W25, S25, M25 then your regex is `[FMSW]\d\d`.  And it should be used to validate the input of the term on the *course entry form*, not anything else.

But, we probably cannot come up with a "regex for regexes" so instead, we leave this required but we don't validate it with a regex.

File: frontend/src/main/components/School/SchoolForm.js [hunk 3]
```diff
@@ -18,14 +18,15 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
 
     // Stryker disable next-line Regex
     const abbrev_regex = /^([a-z._])+$/;
+    const term_regex = /[WSMF]\d\d/;
```
Comment (pconrad) [inline frontend/src/main/components/School/SchoolForm.js]:
```suggestion
```

Remove this; see below.

File: frontend/src/main/components/School/SchoolForm.js [hunk 4]
```diff
@@ -18,14 +18,15 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
...
 
     return (
 
         <Form onSubmit={handleSubmit(submitAction)}>
             <Row>
                 <Col>
                     <Form.Group className="mb-3" >
-                        <Form.Label htmlFor="abbrev">Abbreviation</Form.Label>
+                        <Form.Label htmlFor="abbrev">Abbreviation(All lowercase)</Form.Label>
```
Comment (pconrad) [inline frontend/src/main/components/School/SchoolForm.js]:
```suggestion
                        <Form.Label htmlFor="abbrev">Abbreviation (all lowercase)</Form.Label>
```

File: frontend/src/main/components/School/SchoolForm.js [hunk 5]
```diff
@@ -74,7 +75,7 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
             <Row>
                 <Col>
                     <Form.Group className="mb-3" >
-                        <Form.Label htmlFor="termDescription">Term Description</Form.Label>
+                        <Form.Label htmlFor="termDescription">Term Description(Quarter, Semester or Session)</Form.Label>
```
Comment (pconrad) [inline frontend/src/main/components/School/SchoolForm.js]:
```suggestion
                        <Form.Label htmlFor="termDescription">Term Description (e.g. Quarter, Semester, Session)</Form.Label>
```

File: frontend/src/tests/components/School/SchoolForm.test.js [hunk 6]
```diff
@@ -62,7 +62,7 @@ describe("SchoolForm tests", () => {
 
         await screen.findByText(/Abbreviation is required./);
         expect(screen.getByText(/Name is required./)).toBeInTheDocument();
-        expect(screen.getByText(/Term Regex is required./)).toBeInTheDocument();
+        expect(screen.getByText(/Term Regex has to include a single character for a Quarter and 2 digits for the year./)).toBeInTheDocument();
```
Comment (pconrad) [inline frontend/src/tests/components/School/SchoolForm.test.js]:
```suggestion
        expect(screen.getByText(/Term Regex is required./)).toBeInTheDocument();
```

File: frontend/src/tests/pages/SchoolCreatePage.test.js [hunk 7]
```diff
@@ -88,7 +88,7 @@ describe("SchoolCreatePage tests", () => {
        
         fireEvent.change(abbrevField, { target: { value: 'abb' } });
         fireEvent.change(nameField, { target: { value: 'name-1' } });
-        fireEvent.change(termRegexField, { target: { value: '[WSMF]\\d\\d' } });
+        fireEvent.change(termRegexField, { target: { value: 'W24' } });
```
Comment (pconrad) [inline frontend/src/tests/pages/SchoolCreatePage.test.js]:
```suggestion
        fireEvent.change(termRegexField, { target: { value: "[WSMF]\\d\\d" } });
```

File: frontend/src/tests/pages/SchoolCreatePage.test.js [hunk 8]
```diff
@@ -101,7 +101,7 @@ describe("SchoolCreatePage tests", () => {
             JSON.stringify({
                 "abbrev": "abb",
                 "name": "name-1",
-                "termRegex": "[WSMF]\\d\\d",
+                "termRegex": "W24",
```
Comment (pconrad) [inline frontend/src/tests/pages/SchoolCreatePage.test.js]:
```suggestion
                "termRegex": "[WSMF]\\d\\d" ,
```

**Review by pconrad — CHANGES_REQUESTED** [review]
It appears that your understanding of the purpose of the "term regex" field is not aligned with the intention of that field.

Here's the idea: UCSB uses terms abbreviations such as F24, W25, S25, and M25.

A regular expression for validating those is '[FWMS]\d\d'.

But Washington State University, which has semesters, uses: Fa24, Sp25, Su25, etc.

A regular expression for that would '(Fa|Sp|Su)\d\d'.

So what we want is guidance for the user that helps them understand that.  The first thing we have to do is make sure *we* are on the same page about that.

I've added some suggestions in the code below, but I also want to ask you to look at the the [CommonsForm](https://github.com/ucsb-cs156/proj-happycows/blob/e7487fca555481a59c9ab47d9583b3d2cc8f4a01/frontend/src/main/components/Commons/CommonsForm.js#L112C25-L125C42) from proj-happycows, and the Tooltips in that form. That's what I'm asking for here.   


Specifically, add `OverlayTrigger, Tooltip` in the import for `react-bootstrap`:

```
import {Button, Form, Row, Col, OverlayTrigger, Tooltip} from "react-bootstrap";
```


Then use code such as this from [here](https://github.com/ucsb-cs156/proj-happycows/blob/e7487fca555481a59c9ab47d9583b3d2cc8f4a01/frontend/src/main/components/Commons/CommonsForm.js#L112C25-L125C42).  Note how the `<OverlayTrigger>` surrounds the  `<Form.Control>` element, and the `<Tooltip>` is inside it.

```
<OverlayTrigger
                            placement="top"
                            overlay={<Tooltip>This is the name farmers will see when joining the game.</Tooltip>}
                            delay='5'
                        >
                            <Form.Control
                                data-testid={`${testid}-name`}
                                id="name"
                                type="text"
                                defaultValue={DefaultVals.name}
                                isInvalid={!!errors.name}
                                {...register("name", {required: "Commons name is required"})}
                            />
                        </OverlayTrigger>
```

For Abbreviation, the tooltip could be something like:
```
Please choose an abbreviation for your school such as ucsb, wsu, uh-hilo, etc, consisting of only lowercase characters and hyphens.
```

For Name:
```
The normal full name by which your school is known, e.g. UC Santa Barbara, U Hawaii - Hilo, Washington State, Oregon State, etc.
```

For Term Regex
```
Enter a regular expression that can be used for the abbreviations for academic terms at your school.  For example, if your terms are F24, W25, S25, M25, you can use [FWSM]\d\d.  To allow anything to be entered, use .* for this field.
```

For term description:
```
Please enter the name of the academic term used at your school.  For example: Semester, Quarter, Session, etc.
```

**Review by JUstISAGE — APPROVED** [review]
lgtm

**Review by Peak1260 — APPROVED** [review]
LGTM

**Review by ylin250 — APPROVED** [review]
LGTM!

File: frontend/src/main/components/School/SchoolForm.js [hunk 9]
```diff
@@ -42,30 +53,45 @@ function SchoolForm({ initialContents, submitAction, buttonLabel = "Create" }) {
...
                     <Form.Group className="mb-3" >
-                        <Form.Label htmlFor="termRegex">Term Regex</Form.Label>
-                        <Form.Control
-                            data-testid="SchoolForm-termRegex"
-                            id="termRegex"
-                            type="text"
-                            isInvalid={Boolean(errors.termRegex)}
-                            {...register("termRegex", { required: true })}
-                        />
+                        <Form.Label htmlFor="termRegex">Term Regex(Example: W24)</Form.Label>
```
Comment (pconrad) [inline frontend/src/main/components/School/SchoolForm.js:78]:
```suggestion
                        <Form.Label htmlFor="termRegex">Term Regex (Example: [WSMF]\d\d)</Form.Label>
```

**Review by pconrad — APPROVED** [review]
Technically, there's still one issue with this PR, but I'm going to let it go because we are running out of time.   If you want to make a very quick PR to fix it before class ends tonight, I'll consider merging it; otherwise it will just have to stay like this and we'll fix it later.

**Pull request merged**